lib/ansible/modules/network/enos/enos_facts.py

```diff
@@ -0,0 +1,497 @@
+      transport: cli
+      timeout: 30
+      authorize: True
+      auth_pass:
+
+---
+- name: Collect all facts from the device
+  enos_facts:
+    gather_subset: all
+    provider: "{{ cli }}"
+
+- name: Collect only the config and default facts
+  enos_facts:
+    gather_subset:
+      - config
+    provider: "{{ cli }}"
+
+- name: Do not collect hardware facts
+  enos_facts:
+    gather_subset:
+      - "!hardware"
+    provider: "{{ cli }}"
+
+'''
+RETURN = '''
+  ansible_net_gather_subset:
+    description: The list of fact subsets collected from the device
+    returned: always
+    type: list
+# default
+  ansible_net_model:
+    description: The model name returned from the Lenovo ENOS device
+    returned: always
+    type: str
+  ansible_net_serialnum:
+    description: The serial number of the Lenovo ENOS device
+    returned: always
+    type: str
+  ansible_net_version:
+    description: The ENOS operating system version running on the remote device
+    returned: always
+    type: str
+  ansible_net_hostname:
+    description: The configured hostname of the device
+    returned: always
+    type: string
+  ansible_net_image:
+    description: Indicates the active image for the device
+    returned: always
+    type: string
+# hardware
+  ansible_net_memfree_mb:
+    description: The available free memory on the remote device in MB
+    returned: when hardware is configured
+    type: int
+  ansible_net_memtotal_mb:
+    description: The total memory on the remote device in MB
+    returned: when hardware is configured
+    type: int
+# config
+  ansible_net_config:
+    description: The current active config from the device
+    returned: when config is configured
+    type: str
+# interfaces
+  ansible_net_all_ipv4_addresses:
+    description: All IPv4 addresses configured on the device
+    returned: when interfaces is configured
+    type: list
+  ansible_net_all_ipv6_addresses:
+    description: All IPv6 addresses configured on the device
+    returned: when interfaces is configured
+    type: list
+  ansible_net_interfaces:
+    description: A hash of all interfaces running on the system.
+      This gives information on description, mac address, mtu, speed,
+      duplex and operstatus
+    returned: when interfaces is configured
+    type: dict
+  ansible_net_neighbors:
+    description: The list of LLDP neighbors from the remote device
+    returned: when interfaces is configured
+    type: dict
+'''
+
+import re
+
+from ansible.module_utils.enos import run_commands, enos_argument_spec, check_args
+from ansible.module_utils._text import to_text
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.six import iteritems
+from ansible.module_utils.six.moves import zip
+
+
+class FactsBase(object):
+
+    COMMANDS = list()
+
+    def __init__(self, module):
+        self.module = module
+        self.facts = dict()
+        self.responses = None
+        self.PERSISTENT_COMMAND_TIMEOUT = 60
```

Comment:
I don't see `PERSISTENT_COMMAND_TIMEOUT` used anywhere. Is this required?

Reply:
Removed. Its not required